script.sh

```diff
@@ -99,6 +99,16 @@ echo '::endgroup::'
 echo '::group:: Running trivy with reviewdog 🐶 ...'
   export REVIEWDOG_GITHUB_API_TOKEN="${INPUT_GITHUB_TOKEN}"
 
+  if [[ -n "${INPUT_FAIL_LEVEL}" ]]; then
+    fail_level="--fail-level=${INPUT_FAIL_LEVEL}"
+  elif [[ "${INPUT_FAIL_ON_ERROR}" = "true" ]]; then
+    # For backward compatibility, default to any if fail-on-error is true
+    # Deprecated
+    fail_level="--fail-level=any"
+  else
+    fail_level=""
+  fi
```

Comment:
I don’t think it makes sense for reviewdog users to include conditional branching based on options.
The reviewdog code seems to include code to take care when fail-on-error is true. You can just pass the two options as is and there will be no problem.

```go
func failLevel(opt *option) reviewdog.FailLevel {
	if opt.failOnError {
		slog.Warn("reviewdog: -fail-on-error is deprecated. Use -fail-level=any, or -fail-level=error for github-[pr-]check reporter instead. See also https://github.com/reviewdog/reviewdog/blob/master/CHANGELOG.md")
		if opt.failLevel == reviewdog.FailLevelDefault {
			switch opt.reporter {
			default:
				return reviewdog.FailLevelAny
			case "github-check", "github-pr-check":
				return reviewdog.FailLevelError
			}
		}
	}
	return opt.failLevel
}
```

https://github.com/reviewdog/reviewdog/pull/1854